<launch>
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="52CB:52CB" />
    <param name="brightness" value="128" />  
    <param name="contrast" value="32" />  
    <param name="saturation" value="0" />
    <param name="sharpness" value="0" />
    <!-- mt9v034只支持mmap -->
    <param name="io_method" value="mmap"/>
    <!-- mt9v034只支持752*480*30 -->
    <param name="bStereo" value="false" /> 
    <param name="image_width" value="752" />
    <param name="image_height" value="480" />
    <param name="framerate" value="30" />  
    <!-- 曝光时间延时(忽略) + 数据传输延时, 单位ns, (在摄像头HBV-1911GS yuyv(不压缩)格式720p条件下约10.072Hz(即传输速率为18564710字节=10.7072*1280*720*2)) -->
    <param name="time_delay_per_byte" value="53.865640777582844" />
    <!-- mt9v034只支持 GREY -->
    <param name="v4l_pixel_format" value="GREY" /> 
    <!-- 输出格式: 需为sensor_msgs/image_encodings.h中之一, openvins代码默认使用mono8  -->
    <param name="publish_image_format" value="mono8" />     
    <param name="focus_auto" value="false" /> 
    <param name="focus_absolute" value="68" /> 
    <param name="exposure_auto_priority" value="true" /> 
    <param name="exposure_absolute" value="10000" /> 
    <param name="white_balance_temperature_auto" value="true" /> 
    <param name="white_balance_temperature" value="4600" /> 
    <param name="gain" value="120" /> 
    <param name="gamma" value="120" /> 
    <param name="hue" value="0" /> 
    <param name="backlight_compensation" value="0" /> 
    <param name="pan_absolute" value="0" /> 
    <param name="tilt_absolute" value="0" /> 
    <param name="zoom_absolute" value="0" /> 
    <!-- 由于解码在imx8mq中时间较长,为提高帧率,接收后在另一个线程解码。TODO：但占用内存资源，后期需要把这部分代码优化及移植到算法代码程序里减少内存使用 -->
    <param name="process_image_in_other_thread" value="false" /> 
    <!-- 缓存数据的有效时间 -->
    <param name="image_valid_time" value="0.2" /> 
    <param name="camera_frame_id" value="mt9v034" />
    <param name="camera_name" value="mt9v034" />
    <!-- 请设置校正系数，只用于发布图像信息，发布的图像未校准 -->
    <param name="camera_info_url" value="file://$(find usb_cam)/launch/mt9v034.yaml"/>

    <!-- 使能imx8 isp, dts配置需要对应mipi端口关闭isi并打开isp -->
    <param name="enable_imx_isp" value="false" /> 
    <!-- 使能imx8 dewarp, 用于校正鱼眼和广角镜头图像中引入的失真 -->
    <param name="enable_imx_dewarp" value="false" /> 
  </node>

  <!-- <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/usb_cam/image_raw"/>
    <param name="autosize" value="true" />
  </node> -->
</launch>
